### ISSUE TITLE
Aggregation table shows old data

### ISSUE THREAD

**Issue by dvmc** [cosmocode/dokuwiki-plugin-struct#568]
Since an upgrade to version 2021-03-17 (from 2020-08-26), the data shown in aggregation tables is not updated properly.

After editing the struct data of a page that is shown in the table, the old data of said page remains in the aggregation table, even after clearing the cache (via `touch conf/local.php`).

As a temporary workaround, any modification to the aggregation table page seems to "refresh" the table, and show the latest values of the referenced page data.

To illustrate this issue, here is an example that I just tried out. There is a schema called `myschema` assigned to `wiki:ttt:**`. The following aggregation table can be found at the page id `wiki:ttt:top`.

```
---- struct table ----
schema: myschema
cols: %title%, *
filter: %pageid% ~ wiki:ttt:*
----
```

Afterwards, multiple sub-pages can be added or modified, like `wiki:ttt:top:test1`, `wiki:ttt:top:test2`, etc. However, the changes do not appear in the aggregation table at `wiki:ttt:top` immediately, even after clearing the cache. We have to make any modification to `wiki:ttt:top`, only then the table is updated with the proper values.

Furthermore, when re-assigning `myschema` to `wiki:ttt:top:**`, then the struct table at the top (or bottom) of `wiki:ttt:top` remains. Again, we have to make any modification on the page for the table to disappear.

Possibly related to #563.

Any pointers on where in the source code to look for this would be very helpful, too.

**Comment by solewniczak** [comment]
I think that the bug is caused because the `Doku_Renderer_metadata` is added to `const BLACKLIST_RENDERER` array in helper.php. In line 121 of syntax/table.php the additional metadata should be passed to make the cache dependant on sqlite database file(struct.sqlite3):
```
if ($format === 'metadata') {
    /** @var Doku_Renderer_metadata $renderer */
    $renderer->meta['plugin']['struct']['hasaggregation'] = $search->getCacheFlag();
}
```
but when the `Doku_Renderer_metadata` is blacklisted, the code is never run and the proper cache dependences are not added. 

I'm not sure why the  `Doku_Renderer_metadata` is blacklisted but when I remove it from the list the cache starts to work correctly:
```
    const BLACKLIST_RENDERER = [
//        'Doku_Renderer_metadata',
        '\renderer_plugin_qc'
    ];

```

**Comment by dvmc** [comment]
Blacklisting the metadata renderer seems to have been introduced in #446. But as far as I remember the update of the aggregation table was working properly after that.

**Comment by annda** [comment]
From what I see the problem is not the blacklist as such, but rather the fact that it is applied when rendering aggregation tables. It was introduced for the output of page data and that is where it should stay. 

I removed the erroneous check in ca94dee. This should fix both this issue and #563

**Comment by dvmc** [comment]
Works! Had to clear the cache with `touch conf/local.php` for all sites to work, but now it's fixed. Thanks a lot!
